api/filter/http/buffer.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter.http;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+
+// The buffer filter is used to stop filter iteration and wait for a fully
+// buffered complete request. This is useful in different situations
+// including protecting some applications from having to deal with partial
+// requests and high network latency.
+message Buffer {
+  // The maximum request size that the filter will before the connection
+  // manager will stop buffering and return a 413 response.
```

Comment:
We probably need to do a pass to add required/optional annotations anyway, but you could start by flagging things REQUIRED when (1) it's wrapped and (2) it's required or we barf.